lib/xenops_server.ml

```diff
@@ -1678,12 +1725,16 @@ and perform_exn ?subtask ?result (op: operation) (t: Xenops_task.task_handle) :
         );
       let atomics = [
         VM_hook_script(id, Xenops_hooks.VM_pre_destroy, Xenops_hooks.reason__suspend);
-      ] @ (atomics_of_operation (VM_shutdown (id, None))) @ [
+      ] @ (atomics_of_operation (VM_shutdown (new_src_id, None))) @ [
           VM_hook_script(id, Xenops_hooks.VM_post_destroy, Xenops_hooks.reason__suspend);
+          VM_remove(new_src_id);
         ] in
       perform_atomics atomics t;
-      VM_DB.signal id
-    | VM_receive_memory (id, memory_limit, s) ->
+      VM_DB.signal id;
```

Comment:
But... `id` no longer exists?

I'm a little worried about the hook scripts as well.

Reply:
true, the VM_DB.signal is redundant.

The `pre_migrate` and `post_migrate` hook should be OK. The pre and post destroy are currently being called with the new src id, but at that point the domain is not called that. But that's why these hook scripts are problematic.